ReactWindows/ReactNative/Common/AccessibilityHelper.cs

```diff
@@ -0,0 +1,258 @@
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Automation;
+using Windows.UI.Xaml.Automation.Peers;
+
+namespace ReactNative.Common
+{
+    /// <summary>
+    /// Accessibility helper methods.
+    /// </summary>
+    public static class AccessibilityHelper
+    {
+        #region ImportantForAccessibility - Provides support for ImportantForAccessibility property implementation.
+
+        /// <summary>
+        /// Initializes the value of AccessibilityView for <paramref name="uiElement"/> and its children,
+        /// based on the ancestor <paramref name="parentUIElement"/> settings. Must be callded
+        /// when an element is initially added to the visula tree.
+        /// </summary>
+        /// <param name="parentUIElement">The element where <paramref name="uiElement"/> is added as child.</param>
+        /// <param name="uiElement">The element being added to the visual tree.</param>
+        public static void InitImportantForAccessibility(UIElement parentUIElement, UIElement uiElement)
+        {
+            var parentImportantForAccessibilityAttached = GetImportantForAccessibilityAttached(parentUIElement);
+            var parentUIElementAutomationPeer = FrameworkElementAutomationPeer.FromElement(parentUIElement);
+            if (parentImportantForAccessibilityAttached == ImportantForAccessibility.Yes ||
+                parentImportantForAccessibilityAttached == ImportantForAccessibility.NoHideDescendants ||
```

Comment:
We repeat this logic in a few spots. Should we put it into a helper?

Reply:
we have few places where the check is like xxx=ImportantForAccessibilityYes || xxx=ImportantForAccessibility.NoHideDescendents where both denote that children should not be visible to Narrator, however we don't expect the number of the || to increase so I think that 2 enum checks don't justify a separate method

---
In reply to: [164942705](https://github.com/Microsoft/react-native-windows/pull/1595#discussion_r164942705) [](ancestors = 164942705)

Reply:
My motivation for introducing a helper here is to make it clear what the intention is of checking for "Yes" and "NoHideDescendants". Right now, you look at that code and you have to think "Why are we checking for Yes and NoHideDescendants? What does that mean?"

Whereas if we put that check into a helper called something like "DoesHideDescendants" we would have a better idea of what that check means.

---
In reply to: [165529425](https://github.com/Microsoft/react-native-windows/pull/1595#discussion_r165529425) [](ancestors = 165529425,164942705)

Reply:
Fixed.

---
In reply to: [165803740](https://github.com/Microsoft/react-native-windows/pull/1595#discussion_r165803740) [](ancestors = 165803740,165529425,164942705)